lib/net/hostlist.js

```diff
@@ -787,23 +787,41 @@ class HostList {
     return out;
   }
 
+  /**
+   * Convert address list to a filtered
+   * array. Only addresses without keys
+   * are included.
+   *
+   * @returns {NetAddress[]}
+   */
+
+  toFilteredArray() {
+    const out = [];
+
+    for (const entry of this.map.values())
+      if (!entry.addr.hasKey())
+        out.push(entry.addr);
+
+    return out;
+  }
+
   /**
    * Add a preferred seed.
    * @param {String} host
    */
 
   addSeed(host) {
-    const ip = IP.fromHostname(host, this.network.port);
+    let ip = IP.fromHostname(host, this.network.port);
```

Comment:
Some work could be saved here since `IP.fromHostname` is potentially called twice which duplicates work. I did this for simplicity's sake, since the `ip.key` branch should be the minority. A `hasKey` function could be helpful here for getting the correct port and then calling `IP.fromHostname` a single time.

Reply:
`IP.fromHostname()` returns a [dumb object](https://github.com/bcoin-org/binet/blob/224bbfac6056e0480351f430486f200ca6bc2c25/lib/ip.js#L544-L551), I wonder if you could just set it quickly after the first call:

```
if (ip.key && ip.port === this.network.port)
  ip.port == this.network.brontidePort;
```

The intent would be "if the port is the fallback port for clearnet, assume no custom port was provided. Since we have a key, we should use the brontide fallback port"

This would only fail if someone specifically had a hostname WITH a key, but also used the default clearnet port. Which would be a stupid edge case... so maybe the best way is what you have... or an update to `binet`

Reply:
The reason why I don't love mutating the `ip` object returned from `IP.fromHostName` is because it builds additional properties on that object that would fall out of sync with the change we made, see https://github.com/bcoin-org/binet/blob/master/lib/ip.js#L542.

Its possible to create a `hasKey` function, I feel like that would be better off in `binet` so that it can be consumed internal to the library as well. Maybe we can update `binet` later and go with this for now? Or just write the `hasKey` function and keep it as a helper in this file until we can move it into `binet.